common/src/NodeTree.h

```diff
@@ -26,7 +26,7 @@
 #include <functional>
 #include <list>
 
-template <typename T, size_t S, typename U, typename Cmp = std::less<U>>
```

Comment:
We might as well get rid of this interface altogether, as there is only one implementation now.

Reply:
Was thinking the same, will do